blenderproc/python/utility/PatternUtility.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+import random
+import cv2
+
+from typing import Tuple
+
+def generate_random_pattern_img(width: int, height: int, n_points: int) -> Tuple[int, int, int, int]:
+    """
+    Generate transparent image with random pattern.
+    :param width, height: width and height of image to be generated.
+    :param n_points: number of white points uniformly placed on image.
+    """
+    pattern_img = np.zeros((height, width, 4), dtype=np.uint8)
+
+    m_width = int(width // np.sqrt(n_points))
+    m_height = int(height // np.sqrt(n_points))
+
+    for i in range(width // m_width):
+        for j in range(height // m_height):
+            x_idx = random.randint(i * m_width, (i + 1) * m_width - 1)
+            y_idx = random.randint(j * m_height, (j + 1) * m_height - 1)
+            pattern_img = cv2.circle(pattern_img, (x_idx, y_idx), 1, (255, 255, 255, 255), -1)
```

Comment:
What is the use of this circle function here? Why is it better than just using the index?

Reply:
If the image is large, coloring one pixel is not going to be distinguishable. With `cv2.circle(image, center_coordinates, radius, color, thickness)` we can adjust the radius and get real dots instead of square pixels.